CodeEdit/Features/AppPreferences/Sections/TextEditingPreferences/TextEditingPreferencesView.swift

```diff
@@ -130,6 +133,54 @@ struct TextEditingPreferencesView: View {
         }
     }
 
+    private var textEncoding: some View {
+        Picker("Default Text Encoding", selection: $prefs.preferences.textEditing.textEncoding) {
+            Group {
+                ForEach(String.Encoding.allUnicodes, id: \.id) { unicode in
+                    Text(unicode.description)
+                        .tag(unicode)
+                }
```

Comment:
I suggest making an array of all groups and iterating over those something like this since it is still very repetitive:

```swift
let allGroups [[String.Encoding]] = [
    .allUnicode,
    .allAscii,
    .allJapanese,
    …
]

ForEach(allGroups) { group in
    ForEach(group) { encoding in
        // `.description` could be removed since it
        // conforms to `CustomStringConvertible`
        Text(encoding.description)
            .tag(encoding.rawValue)
    }
    if group != allGroups.last {
        Divider() 
    }
}
```